packages/clay-list/src/List.tsx

```diff
@@ -41,7 +49,19 @@ const ClayList: React.FunctionComponent<IProps> & {
 				'show-quick-actions-on-hover': showQuickActionsOnHover,
 			})}
 		>
-			{children}
+			{children &&
+				React.Children.map(children, (child) => {
+					warning(
```

Comment:
So `warning` has an `if (__DEV__)` check inside it (great), but it would be nice to skip the entire `React.Children.map` call when `process.env.NODE_ENV !== 'development`.